pydoctor/epydoc/markup/epytext.py

```diff
@@ -1352,69 +1356,107 @@ def to_stan(self, docstring_linker: DocstringLinker) -> Tag:
+        # Process the children first.
+        variables: List[Node] = []
+        for c in tree.children:
+            variables.extend(self._to_node(c, seclevel))
 
         # Perform the approriate action for the DOM tree type.
         if tree.tag == 'para':
-            if tree.attribs.get('inline'):
-                return variables
-            else:
-                return tags.p(*variables)
+            # we yield a paragraph node even if tree.attribs.get('inline') is True because
+            # the choice to render the <p> tags is handled in _PydoctorHTMLTranslator.should_be_compact_paragraph(), not here anymore
+            yield set_node_attributes(paragraph('', ''), document=self._document, children=variables)
         elif tree.tag == 'code':
-            return tags.code(*variables)
+            yield set_node_attributes(literal('', ''), document=self._document, children=variables)
         elif tree.tag == 'uri':
-            return tags.a(variables[0], href=variables[1], target='_top')
+            _label, _target = variables[0], variables[1]
+            yield set_node_attributes(reference(
+                    '', internal=False, refuri=_target), document=self._document, children=_label.children)
+        
         elif tree.tag == 'link':
-            label, target = variables
-            lineno = int(cast(Element, tree.children[1]).attribs['lineno'])
-            return linker.link_xref(target, label, lineno)
+            _label, _target = variables
+            assert isinstance(_target, Text)
+            assert isinstance(_label, inline)
+
+            args = {}
+            if _target.astext() != _label.astext():
+                args['refuri']=_target.astext()
+            
+            # Figure the line number to warn on precise lines. 
+            # This is needed only for links currently.
+            lineno: Optional[int] = None
+            elem = tree.children[1]
+            if isinstance(elem, Element):
+                lineno = int(elem.attribs['lineno'])
+
+            yield set_node_attributes(title_reference(
+                   '', '', **args), document=self._document, lineno=lineno, children=_label.children)
+
+        elif tree.tag in ('name',):
+            yield set_node_attributes(inline('', ''), document=self._document, children=variables)
+            # yield set_node_attributes(Text(' '.join(node2stan.gettext(variables))))
         elif tree.tag == 'target':
             value, = variables
-            return value
+            yield set_node_attributes(Text(value), document=self._document)
+
         elif tree.tag == 'italic':
-            return tags.i(*variables)
+            yield set_node_attributes(emphasis('', ''), document=self._document, children=variables)
         elif tree.tag == 'math':
-            return tags.i(*variables, class_='math')
+            node = set_node_attributes(math('', ''), document=self._document, children=variables)
+            node.set_class('math')
```

Comment:
This triggers a deprecation warning: 

`DeprecationWarning: docutils.nodes.Element.set_class deprecated; append to Element['classes'] list attribute directly`

Reply:
Replaced with `node['classes'].append('math')`